src/FffPolygonGenerator.cpp

```diff
@@ -607,6 +637,12 @@ void FffPolygonGenerator::processInfillMesh(SliceDataStorage& storage, const siz
             layer.parts.back().outline = part;
             layer.parts.back().boundaryBox.calculate(part);
         }
+        // TODO: reclose surface mode polygons which weren't cut up into broken polylines; and make LayerParts for them
```

Comment:
I guess this would still be a TODO for the future. In most cases your infill mesh will be cut up into segments, but some infill meshes could consist of smaller polygons which are fully enclosed inside the infill. Such polygons are now pre-cut, meaning that their start point is determined randomly, rather than being determines by the path order optimizer.

I considered this bug to be too niche to solve in this PR. Tell me if you think it should be solved anyway.